lib/facter/facts/sles/os/distro/codename.rb

```diff
@@ -6,13 +6,16 @@ module Os
       module Distro
         class Codename
           FACT_NAME = 'os.distro.codename'
-          ALIASES = 'lsbdistcodename'
 
           def call_the_resolver
-            fact_value = Facter::Resolvers::LsbRelease.resolve(:codename)
+            fact_value = Facter::Resolvers::SpecificReleaseFile.resolve(
+              :release, release_file: '/etc/os-release'
+            ).match(/.*release.*(\(.*\)).*/)
```

Comment:
Use `Facter::Resolvers::OsRelease.resolve(:pretty_name)`

Reply:
fact_value can be nil if VERSION_CODENAME is not resent in `/etc/os-release`